src/allmydata/storage/http_client.py

```diff
@@ -282,3 +279,23 @@ def read_share_chunk(
             raise ClientException(
                 response.code,
             )
+
+    @inlineCallbacks
+    def list_shares(self, storage_index):  # type: (bytes,) -> Deferred[Set[int]]
+        """
+        Return the set of shares for a given storage index.
+        """
+        url = self._client._url(
+            "/v1/immutable/{}/shares".format(_encode_si(storage_index))
+        )
+        response = yield self._client._request(
+            "GET",
+            url,
+        )
+        if response.code == http.OK:
+            body = yield response.content()
+            returnValue(set(loads(body)))
```

Comment:
After https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3861 this code (or some orthogonal abstraction?) will look at `Content-Type` of the response and choose decoder appropriately, I guess.  And maybe also something about applying validation based on the schema?

Reply:
Oh right I had a utility for this didn't I. So I'll use that so there's only one place to fix later.